libsolidity/formal/SMTEncoder.cpp

```diff
@@ -2690,32 +2705,27 @@ pair<FunctionDefinition const*, ContractDefinition const*> SMTEncoder::functionC
 		calledExpr = innermostTuple(*calledExpr);
 	}
 
-	auto resolveVirtual = [&](auto const* _ref) -> pair<FunctionDefinition const*, ContractDefinition const*> {
+	auto resolveVirtual = [&](auto const* _ref) -> FunctionDefinition const* {
 		VirtualLookup lookup = *_ref->annotation().requiredLookup;
-		solAssert(_contract || lookup == VirtualLookup::Static, "No contract context provided for function lookup resolution!");
+		solAssert(_contextContract || lookup == VirtualLookup::Static, "No contract context provided for function lookup resolution!");
 		auto funDef = dynamic_cast<FunctionDefinition const*>(_ref->annotation().referencedDeclaration);
 		if (!funDef)
-			return {funDef, _contract};
-		ContractDefinition const* contextContract = nullptr;
+			return funDef;
 		switch (lookup)
 		{
 		case VirtualLookup::Virtual:
-			funDef = &funDef->resolveVirtual(*_contract);
-			contextContract = _contract;
-			break;
+			return &funDef->resolveVirtual(*_contextContract);
```

Comment:
```suggestion
			return &(funDef->resolveVirtual(*_contextContract));
```
Do you think it helps with readability?

Reply:
I prefer the former, but also fine with changing